File: 2_all_words_once/src/all_words_once.cs
using System;
using System.IO;
using System.Linq;

class Program {
    static (string WordList, int Count) remove_dup_words(string inputString) {
        var wordlist = inputString.Split()
                                  .Where(s => !string.IsNullOrWhiteSpace(s))
                                  .ToHashSet();

        return (WordList: string.Join(" ", wordlist), Count: wordlist.Count);
    }


    static void Main(string[] args) {
        if (args.Length != 2) {
            Console.Error.WriteLine("Parameters: filename repeat_of_test");
            System.Environment.Exit(1);
        }

        var filename = args[0];
        var repeat = Int32.Parse(args[1]);

        var contents = File.ReadAllText(filename);
        (string WordList, int Count) result = (null, 0);

        for (int i = 0; i < repeat; i++) {
            result = remove_dup_words(contents);
        }

        Console.WriteLine($"{result.WordList}");
        Console.Error.WriteLine("CSharp");
    }
}

File: 2_all_words_once/src/all_words_once.rs
use std::collections::HashSet;

fn remove_dup_words(s: &str) -> String {
    let mut wordlist: HashSet<&str> = HashSet::new();

    let v: Vec<&str> = s.split_whitespace()
                        .filter(|w| wordlist.insert(w))
                        .collect();
    v.join(" ")
}

fn main() {
    let args: Vec<String> = std::env::args().collect();
    if args.len() != 3 {
        eprintln!("Parameters: filename repeat_of_test");
        std::process::exit(1);
    }

    let filename = &args[1];
    let repeat: i32 = args[2].parse().unwrap();

    let contents = std::fs::read_to_string(filename).unwrap();
    let mut res = String::new();

    for _ in 0..repeat {
        res = remove_dup_words(&contents);
    }
    println!("{}", res);
    eprintln!("Rust");
}
